### ISSUE TITLE
Use promote rule to generate doc/sphinx/README.rst

### ISSUE THREAD

**Issue by Zimmi48** [coq/coq#8442]
`doc/sphinx/README.rst` contains the following comment:

```rst
..
   README.rst is auto-generated from README.template.rst and the coqrst docs;
   use ``doc/tools/coqrst/regen_readme.py`` to rebuild it.
```

However, people usually miss this so the best way to avoid any sync problem would be to add a test to the test-suite like we did for some extracted ML files.

**Comment by JasonGross** [comment]
I think `doc/sphinx/README.rst` should also be made read-only, if it is not already.

**Comment by JasonGross** [comment]
`chmod +w; modify; chmod -w`
The important thing is that the checked-in version is read-only, so there's an extra barrier to manual modification.  I noticed that CompCert (or maybe it was VST) does this, and found it useful.

**Comment by SkySkimmer** [comment]
This seems more like a lint thing than a test suite thing.

**Comment by Zimmi48** [comment]
@JasonGross Why not? Should we do the same for the above mentioned extracted ML files? @SkySkimmer Sure, any solution is good as long as it ends up being run by CI.

**Comment by JasonGross** [comment]
@Zimmi48 Yes, I'd suggest doing this for any file for which it would be a mistake to commit an edit-by-hand.

**Comment by Zimmi48** [comment]
This seems like something that should be doable with dune, without going in the test-suite. The reason to put the test of the above mentioned extracted ML files in the test-suite is to avoid creating a too long dependency chain to build said ML files. This concern does not apply in the case of the README.

**Comment by ejgallego** [comment]
Bug added to #8052

**Comment by Zimmi48** [comment]
This has been addressed in #9568.
